security.md

```diff
@@ -0,0 +1,42 @@
+
+## For package maintainers: SSWG Security requirements
+
+Packages that are listed on the [SSWG's package index](https://swift.org/server/#projects) are required to follow the following guidelines around security.
+
+When it comes to security vulnerabilities it is key to make sure that somebody who discovers a vulnerability in your package can learn how to report very quickly. As the package author, you know best where to put important information around your software. In general it is probably the `README.md` file in the root of your repository on GitHub, GitLab, or similar. Below "main information page" will refer to the `README.md` or a similar easy-to-find resource with information about your project.
+
+The key requirements are:
+
+- Have a clear point of contact (eg. email address) for security vulnerability reporting listed on your main information page.
+- The point of contact for security vulnerabilities should be private. That means it is only accessible to your maintainers and _not_ publicly accessible on the internet. It is also good practise to describe exactly who can access the information next to the point of contact.
+- Either on the main information page or a page linked from it, describe as easily and precisely as possible how the security process of your package works. This includes where and how to report as well as expected timelines for a fix.
+- The goal is to make it as easy as possible for a user to report a security vulnerability to you. Try to remove all ambiguity and difficulty from reporting vulnerabilities. In many cases you will rely on somebody's spare time to report a vulnerability to you.
+
+## For Package maintainers: What to do if there is a security vulnerability
+
+The discovery of a security vulnerability in your code can be daunting but it is part of the life of any software developer. So first of all, don't stress. If you need any help at any step of the way, please feel free to contact any of the SSWG members or the group as a whole at [sswg-security-reports@forums.swift.org](mailto:sswg-security-reports@forums.swift.org).
+
+Here is a step-by-step guide on what to do with regards to the SSWG.
+
+1. The SSWG is only a secondary point of contact which can broadcast information about vulnerabilities. The best way to start is to start addressing the vulnerability according the the security process of your own package.
```

Comment:
I'm not sure what the entire role of the SSWG is here, it seems to be a few different things: keeping "in the loop", providing guidelines for package authors/maintainers and acting as a fallback for when packages don't have enough information on how to deal with vulnerabilities. Is there anything else?

re: the first part, what does the SSWG do with the information? Previously there was a note about packages failing to appropriately deal with issues having their status lowered or removed from the package index. Is that intentionally dropped?

Reply:
> I'm not sure what the entire role of the SSWG is here, it seems to be a few different things: keeping "in the loop", providing guidelines for package authors/maintainers and acting as a fallback for when packages don't have enough information on how to deal with vulnerabilities. Is there anything else?
> 
> re: the first part, what does the SSWG do with the information? Previously there was a note about packages failing to appropriately deal with issues having their status lowered or removed from the package index. Is that intentionally dropped?

I think that's pretty much it.


> re: the first part, what does the SSWG do with the information?

- verify that processes work and are followed
- being aware that something potentially dangerous is out there, so in certain cases we may want to commit resources to make an extra effort to contact known adopters



>  Previously there was a note about packages failing to appropriately deal with issues having their status lowered or removed from the package index. Is that intentionally dropped?

I thought it sounded like a threat and maybe a bit over the top. I thought that if a package doesn't follow the guidelines it's kinda obvious that it may be demoted and doesn't need to be called out here. But maybe it's good to explicitly state it.

Reply:
Cool, I think it might be worth explicitly stating the role of the SSWG in the section for maintainers. If anything it's useful to make it more obvious that they are there to offer guidance and help when necessary.

---

> I thought it sounded like a threat and maybe a bit over the top. I thought that if a package doesn't follow the guidelines it's kinda obvious that it may be demoted and doesn't need to be called out here. But maybe it's good to explicitly state it.

It _is_ a threat though isn't it? No strong feelings whether it's over the top or not.

Reply:
I think it isn't necessary in the document, that's something better kept outlined in the Incubation Process. I feel this doc should be entirely focused on what @weissi said: A place to point to for different personas on how to move forward with vulnerabilities